src/PolyType/Abstractions/TypeShapeProvider.cs

```diff
@@ -43,4 +43,31 @@ public static ITypeShape<T> Resolve<T>(this ITypeShapeProvider shapeProvider)
     /// <exception cref="NotSupportedException"><paramref name="shapeProvider"/> does not support <paramref name="type"/>.</exception>
     public static ITypeShape Resolve(this ITypeShapeProvider shapeProvider, Type type)
         => shapeProvider.GetShape(type) ?? throw new NotSupportedException($"The shape provider '{shapeProvider.GetType()}' does not support type '{type}'.");
+
+    /// <summary>
+    /// Creates a new <see cref="ITypeShapeProvider"/> that returns the first shape found
+    /// from iterating over a given list of providers.
+    /// </summary>
+    /// <param name="providers">The sequence of providers to solicit for type shapes.</param>
+    /// <returns>The aggregating <see cref="ITypeShapeProvider"/>.</returns>
+    public static ITypeShapeProvider Combine(params ReadOnlySpan<ITypeShapeProvider> providers)
+    {
+        return new AggregatingTypeShapeProvider(providers.ToArray());
+    }
+
+    private sealed class AggregatingTypeShapeProvider(ITypeShapeProvider[] providers) : ITypeShapeProvider
```

Comment:
One issue we quickly encountered with STJ is that there was a need for users to be able introspect and/or modify the chain of resolvers/providers. A quick way to achieve this is making the aggregating provider public and also have it implement `IReadOnlyList<ITypeShapeProvider>`.

Reply:
To make it _modifiable_ would require exposing a property too, that is either mutable or references a mutable collection. Mutation opens up thread-safety concerns, which I tend to want to avoid. 

Here's a possible proposal then:
1. Remove the `Combine` method.
2. Expose a public aggregating provider with a public ctor, and a public property for an immutable collection type, but offer a setter so they can replace it in an atomic way.

I would personally prefer that the provider remains immutable, and let the user create a new one and swap out the instance in their own code and deal with the thread-safety ramifications there.
No concerns with allowing them to enumerate the sub-providers though.

Reply:
I'd prefer keeping it immutable. If modifications become necessary you can always create a fresh aggregated instance.